arch/risc-v/src/common/riscv_common_irq.c

```diff
@@ -0,0 +1,74 @@
+/****************************************************************************
+ * arch/risc-v/src/common/riscv_common_irq.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <nuttx/irq.h>
+#include <sys/types.h>
+
+#include "riscv_internal.h"
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+void riscv_common_irq_attach(void)
+{
+  irq_attach(RISCV_IRQ_IAMISALIGNED, riscv_exception, NULL);
+  irq_attach(RISCV_IRQ_IAFAULT, riscv_exception, NULL);
+  irq_attach(RISCV_IRQ_IINSTRUCTION, riscv_exception, NULL);
+  irq_attach(RISCV_IRQ_BPOINT, riscv_exception, NULL);
+  irq_attach(RISCV_IRQ_LAMISALIGNED, riscv_exception, NULL);
+  irq_attach(RISCV_IRQ_LAFAULT, riscv_exception, NULL);
+  irq_attach(RISCV_IRQ_SAMISALIGNED, riscv_exception, NULL);
+  irq_attach(RISCV_IRQ_SAFAULT, riscv_exception, NULL);
+
+  /* Attach the ecall interrupt handler */
+
+#ifndef CONFIG_BUILD_FLAT
+  irq_attach(RISCV_IRQ_ECALLU, riscv_swint, NULL);
+#else
+  irq_attach(RISCV_IRQ_ECALLU, riscv_exception, NULL);
+#endif
+
+  irq_attach(RISCV_IRQ_ECALLS, riscv_exception, NULL);
+  irq_attach(RISCV_IRQ_ECALLH, riscv_exception, NULL);
+
+#ifndef CONFIG_ARCH_USE_S_MODE
+  irq_attach(RISCV_IRQ_ECALLM, riscv_swint, NULL);
```

Comment:
@pussuw Could you take a look ? Is it OK for S mode ?

Reply:
It should be, ECALLS and ECALLM are handled by SBI.